src/KafkaSchemaRegistryApiApiClient.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+namespace Jobcloud\Kafka\SchemaRegistryClient;
+
+use Buzz\Exception\RequestException;
+use Exception;
+use Jobcloud\Kafka\SchemaRegistryClient\Exception\SchemaNotFoundException;
+use Jobcloud\Kafka\SchemaRegistryClient\Exception\SubjectNotFoundException;
+
+class KafkaSchemaRegistryApiApiClient implements KafkaSchemaRegistryApiClientInterface
+{
+    /**
+     * @var HttpClientInterface
+     */
+    private $httpClient;
+
+    /**
+     * KafkaSchemaRegistryApi constructor.
+     * @param HttpClientInterface $httpClient
+     */
+    public function __construct(HttpClientInterface $httpClient)
+    {
+        $this->httpClient = $httpClient;
+    }
+
+    /**
+     * @return array
+     */
+    public function getSubjects(): array
+    {
+        return $this->httpClient->call('GET', 'subjects') ?? [];
+    }
+
+    /**
+     * @param string $subjectName
+     * @return array
+     */
+    public function getAllSubjectVersions(string $subjectName): array
+    {
+        return $this->httpClient->call('GET', sprintf('subjects/%s/versions', $subjectName)) ?? [];
+    }
+
+    /**
+     * @param string $subjectName
+     * @param string $version
+     * @return array
+     */
+    public function getSchemaByVersion(string $subjectName, string $version = self::VERSION_LATEST): array
+    {
+        return $this
+                ->httpClient
+                ->call('GET', sprintf('subjects/%s/versions/%s', $subjectName, $version)) ?? [];
+    }
+
+    /**
+     * @param string $subjectName
+     * @param string $version
+     * @return string
+     */
+    public function getSchemaDefinitionByVersion(string $subjectName, string $version = self::VERSION_LATEST): string
+    {
+        return json_encode($this
+            ->httpClient
+            ->call('GET', sprintf('subjects/%s/versions/%s/schema', $subjectName, $version)), JSON_THROW_ON_ERROR);
+    }
+
+    /**
+     * @param string $subjectName
+     * @param string $version
+     * @return bool
+     */
+    public function deleteSchemaVersion(string $subjectName, string $version = self::VERSION_LATEST): bool
+    {
+        $this->httpClient->call('DELETE', sprintf('subjects/%s/versions/%s', $subjectName, $version));
```

Comment:
```suggestion
        return $this->httpClient->call('DELETE', sprintf('subjects/%s/versions/%s', $subjectName, $version));
```
probably makes more sense, no?

Reply:
idk

Reply:
done